### PR TITLE
Adds quirk setting to skip removing children in unmount

### PR DESCRIPTION
**Pull request by rozele**
### Type of Change
_Erase all that don't apply._
- New feature (non-breaking change which adds functionality)

### Why
The `removeChildren` option in `DropView` can be used to conditionally recurse the Yoga and native view tree to clean up each node. This would be required if we wanted to pool or otherwise recycle Yoga nodes or XAML views, but we do not do this today. For large UI transitions involving hundreds or thousands of nodes, this recursion can be quite costly so we should evaluate whether it's truly needed.

### What
While we could just explicitly set this to false, my concern is that this could create potential memory leaks, and its better to default to the pre-existing behavior and opt-in for experimentation. Thus the behavior is hidden behind a QuirkSetting.

In addition to adding the QuirkSetting, this change also wires up a change event on QuirkSettings that can be subscribed to in PaperUIManagerModule so we can get fresh quirk settings when they change. While quirk settings are generally set before starting the ReactInstance, if someone wanted to A/B test this functionality and hide it behind a feature flag, it's possible that they might want to change the QuirkSetting at runtime.

## Testing
Added a new QuirkSetting menu to Playground-win32, and tested out checking the box and unchecking the box at runtime. I confirmed that the QuirkSettingsChanged callback fires as expected, the values are refreshed and the PaperUIManagerModule uses the correct value when changed.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10590)

### CODE REVIEW

**Review by jonthysell — APPROVED** [review]

File: vnext/Microsoft.ReactNative/QuirkSettings.idl [hunk 1]
```diff
@@ -46,5 +46,11 @@ namespace Microsoft.ReactNative
       "`inactive` tracks the [Window.Activated Event](https://docs.microsoft.com/uwp/api/windows.ui.core.corewindow.activated) when the window is deactivated.")
     DOC_DEFAULT("false")
     static void SetMapWindowDeactivatedToAppStateInactive(ReactInstanceSettings settings, Boolean value);
+
+    DOC_STRING(
+      "By default `react-native-windows` will recursively remove XAML and Yoga children from parent nodes on unmount. "
+      "Setting this to true enables `react-native-windows` to skip this recursive tree cleanup."
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/QuirkSettings.idl:52]:
```suggestion
      "Setting this to true enables `react-native-windows` to skip this recursive tree cleanup.")
```

**Comment by rozele** [comment]
@jonthysell thanks for the fix. I've converted this to draft because we're going to validate the impact of this in Messenger before landing here.

File: vnext/Microsoft.ReactNative/QuirkSettings.cpp [hunk 2]
```diff
@@ -23,6 +25,7 @@ winrt::Microsoft::ReactNative::ReactPropertyId<bool> MatchAndroidAndIOSStretchBe
     winrt::Microsoft::ReactNative::ReactPropertyBag properties,
     bool value) noexcept {
   properties.Set(MatchAndroidAndIOSStretchBehaviorProperty(), value);
+  s_quirkSettingsChangedEvent(settings.Properties(), MatchAndroidAndIOSStretchBehaviorProperty().Handle());
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/QuirkSettings.cpp:28]:
```suggestion
  s_quirkSettingsChangedEvent(properties.Handle(), MatchAndroidAndIOSStretchBehaviorProperty().Handle());
```

**Comment by rozele** [comment]
No net improvement from this experiment.

**Pull request closed without merging**